src/types.rs

```diff
@@ -488,6 +490,146 @@ impl<T> Line<T>
 pub struct LineString<T>(pub Vec<Point<T>>) where T: Float;
 
 impl<T: Float> LineString<T> {
+    /// Create a new `LineString` from a `Point`s vector.
+    ///
+    /// # Errors
+    ///
+    /// This function returns an `Err` if `points` has fewer than two `Point`s.
+    ///
+    /// # Examples
+    ///
+    /// Creating a `LineString`:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.), Point::new(2., -10.)];
+    /// assert!(LineString::new(points).is_ok());
+    /// ```
+    ///
+    /// Attempting to create a `LineString` with fewer than two `Point`s
+    /// results in an error:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.)];
+    /// assert!(LineString::new(points).is_err());
+    /// ```
+    pub fn new(points: Vec<Point<T>>) -> Result<Self, LineStringError<T>> {
+        if points.len() > 1 {
+            Ok(LineString(points))
+        } else {
+            Err(LineStringError(points))
+        }
+    }
+
+    /// Create a new `LineString` from a `Point`s vector. Only use this
+    /// function if you know _for certain_ that `points` has at least two
+    /// `Point`s.
+    ///
+    /// # Safety
+    ///
+    /// This function is `unsafe` because (soon) some `LineString` internals
+    /// ignore bounds checking when indexing as a performance improvement.
+    ///
+    /// # Examples
+    ///
+    /// Creating a `LineString`:
+    ///
+    /// ```
+    /// use geo::{LineString, Point};
+    ///
+    /// let points = vec![Point::new(6., 14.), Point::new(2., -10.)];
+    /// let line_string = unsafe { LineString::new_unchecked(points) };
+    /// ```
+    pub unsafe fn new_unchecked(points: Vec<Point<T>>) -> Self {
+        debug_assert!(points.len() > 1);
```

Comment:
Hmm... I'm not sure if this `debug_assert!()` would be valid here. If a developer is already using `unsafe` then that's a promise by him/her that the thing they're passing in is valid and doesn't need to be checked. 

Also, doesn't naming the function `new_unchecked()` and then checking it seem like a bit of an oxymoron? :stuck_out_tongue_winking_eye:

Reply:
Yeah, you're right. For comparison, the `unsafe` function [`CString::from_vec_unchecked`](https://github.com/rust-lang/rust/blob/23aaeb573b626a51af9ecc97680663153e4ab2b0/src/libstd/ffi/c_str.rs#L272-L276) could very easily do a `debug_assert` to make sure that the `Vec` that is passed in does _not_ have a trailing null byte, but it doesn't.